clients/src/main/resources/common/message/StopReplicaRequest.json

```diff
@@ -37,12 +41,24 @@
       { "name": "PartitionIndex", "type": "int32", "versions": "0",
         "about": "The partition index." }
     ]},
-    { "name": "Topics", "type": "[]StopReplicaTopic", "versions": "1+",
+    { "name": "Topics", "type": "[]StopReplicaTopicV1", "versions": "1-2",
       "about": "The topics to stop.", "fields": [
-      { "name": "Name", "type": "string", "versions": "1+", "entityType": "topicName",
+      { "name": "Name", "type": "string", "versions": "1-2", "entityType": "topicName",
         "about": "The topic name." },
-      { "name": "PartitionIndexes", "type": "[]int32", "versions": "1+",
+      { "name": "PartitionIndexes", "type": "[]int32", "versions": "1-2",
         "about": "The partition indexes." }
+    ]},
+    { "name": "TopicStates", "type": "[]StopReplicaTopicState", "versions": "3+",
+      "about": "Each topic.", "fields": [
+      { "name": "TopicName", "type": "string", "versions": "3+", "entityType": "topicName",
+        "about": "The topic name." },
+      { "name": "PartitionStates", "type": "[]StopReplicaPartitionState", "versions": "3+",
```

Comment:
It's a bit annoying that the `DeletePartitions` flag is set at the top level. It would be nice to move that into the partition state. A KIP for another day I guess...

Reply:
Yes. I totally agree with you. It would avoid us to always send out two StopReplicaRequests. I almost did it directly but I thought that it would stretch the KIP too far perhaps.  What do you think? Shall we do it directly? Anyway, we should do it before 2.6 is released in order to avoid bumping the API twice. I am fine with both ways.

Reply:
I'd suggest sending a message to the vote thread and ask what other think about the change. It seems like a no-brainer improvement to me, so maybe we don't need the overhead of doing another KIP.

Reply:
Done. I have also implemented it directly ;)